### PR TITLE
Kernel+AK+LibC: Make the kernel build with Clang

### PR DESCRIPTION
**Pull request by BertalanD**
This PR fixes the small things that made the kernel not build with Clang.

Note that some warnings (like `cast-align`) still have to be turned off manually, as fixing all issues related to it would take an unreasonably large effort. See #8329, which describes why GCC does not catch these bugs.

### CODE REVIEW

**Comment by BertalanD** [comment]
Probably will need to change the LibC `_Alignas` stuff, because it is incompatible with C99, which some ports use.

File: Userland/Libraries/LibC/assert.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 __BEGIN_DECLS
 
 #ifdef DEBUG
-[[noreturn]] void __assertion_failed(const char* msg);
+__attribute__((noreturn)) void __assertion_failed(const char* msg);
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibC/assert.h:14]: weirdly enough [this](en.cppreference.com/w/cpp/language/attributes) says that `[[noreturn]]` is a standardised attribute

Comment (BertalanD) [reply Userland/Libraries/LibC/assert.h:14]: Yes, it is standardized in C++.

However, it's [not yet](https://en.cppreference.com/w/c/language/attributes) part of the C standard. Most ports written in C did not care as GCC already has this as a "language extension". But technically speaking, that's not valid C code. Clang's `compiler-rt`, however, expects the C standard library to be standards-conformant, and failed to compile.

**Comment by BertalanD** [comment]
@alimpfard I decided not to mess with launder in `Variant`, as things seem to work as they are. I realized that we don't need to worry about UB, as we currently aren't storing const or reference types in them. The possible UB in those cases could probably best fixed by just not allowing to use these types.

Furthermore, I did a little digging into the actual standard, and it says that the restrictions around aliasing don't matter for member variables.

**Review by alimpfard — COMMENTED** [review]
re "Kernel: Fix remaining warnings"
Perhaps this could use a more descriptive commit message, otherwise LGTM

**Comment by BertalanD** [comment]
Changed the title. We should probably wait to see if it works correctly on x86_64.

**Comment by alimpfard** [comment]
Alright, we'll merge this after the whole x86_64 thing has settled a bit then

**Comment by BertalanD** [comment]
CI succeeded and my Clang image ran the on-target tests without crashing, so I think this PR is ready to be merged.

**Pull request merged**